kafka-ui-react-app/src/components/KsqlDb/QueryModal/QueryModal.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { FC, useCallback, useState } from 'react';
+import { yupResolver } from '@hookform/resolvers/yup';
+import SQLEditor from 'components/common/SQLEditor/SQLEditor';
+import yup from 'lib/yupExtended';
+import { Controller, useForm } from 'react-hook-form';
+import { KsqlCommand } from 'generated-sources';
+import { ksqlDbApiClient } from 'redux/actions/thunks/ksqlDb';
+import Tabs from 'components/common/Tabs/Tabs';
+import JSONEditor from 'components/common/JSONEditor/JSONEditor';
+import { isEmpty } from 'lodash';
+
+export interface QueryModalProps {
+  isOpen?: boolean;
+  onSubmit?(): void;
+  onCancel(): void;
+  clusterName: string;
+  isConfirming?: boolean;
+}
+
+type FormValues = KsqlCommand;
+
+const validationSchema = yup.object({
+  ksql: yup.string().trim().required(),
+});
+
+const TABS_INITIAL_PAGE = 0;
+const RESULT_INITIAL_VALUE = {};
+
+const QueryModal: FC<QueryModalProps> = ({
+  isOpen,
+  clusterName,
+  onCancel,
+  onSubmit,
+  isConfirming = false,
+}) => {
+  if (!isOpen) return null;
+
+  const [result, setResult] = useState(RESULT_INITIAL_VALUE);
+
+  const {
+    handleSubmit,
+    control,
+    formState: { isSubmitting },
+  } = useForm<FormValues>({
+    mode: 'onTouched',
+    resolver: yupResolver(validationSchema),
+    defaultValues: {
+      ksql: '',
```

Comment:
there is second non required param `additionalProperties`.  Don't forget to add it.